src/quantity.rs

```diff
@@ -508,11 +508,14 @@ macro_rules! quantity {
                     fn from_str(s: &str) -> Result<Self, Self::Err> {
                         let mut parts = s.splitn(2, ' ');
                         let value = parts.next().unwrap();
-                        let abbr = parts.next().ok_or(NoSeparator)?;
+                        let unit = parts.next().ok_or(NoSeparator)?;
                         let value = value.parse::<V>().map_err(|_| ValueParseError)?;
 
-                        match abbr.trim() {
+                        #[allow(unreachable_patterns)]
+                        match unit.trim() {
                             $($abbreviation => Ok(Self::new::<super::super::$unit>(value)),)+
+                            $($singular => Ok(Self::new::<super::super::$unit>(value)),)+
+                            $($plural => Ok(Self::new::<super::super::$unit>(value)),)+
                             _ => Err(UnknownUnit),
                         }
```

Comment:
The current code can be simplified:

```rust
                        match unit.trim() {
                            $($abbreviation | $singular | $plural => Ok(Self::new::<super::super::$unit>(value)),)+
                            _ => Err(UnknownUnit),
                        }
```

We could also consider only matching singular when `value.is_one()` and plural when not. This would require separate matches as written currently in order to fit the match guards. If we do end up going this route then only one macro iteration `$(...)+` is needed around all three lines instead of one on each line.

Reply:
> We could also consider only matching singular when value.is_one() and plural when not. 

As @bheisler said above, being strict in this sense seems to provide little benefit especially when parsing user input. I would suggest erring on the side of lenience here following the parsing adage "be lenient in what you accept, be strict in what you generate".

Reply:
I've updated it to simplify the code, as suggested.